service/collector/vpc.go

```diff
@@ -46,59 +46,91 @@ var (
 func (c *Collector) collectVPCs(ch chan<- prometheus.Metric) {
 	c.logger.Log("level", "debug", "message", "collecting metrics for vpcs")
 
-	i := &ec2.DescribeVpcsInput{
-		Filters: []*ec2.Filter{
+	f := func(filters []*ec2.Filter) {
+		i := &ec2.DescribeVpcsInput{
+			Filters: filters,
+		}
+		o, err := c.awsClients.EC2.DescribeVpcs(i)
+		if err != nil {
+			c.logger.Log("level", "error", "message", "could not list vpcs", "stack", fmt.Sprintf("%#v", err))
+		}
+
+		for _, vpc := range o.Vpcs {
+			cluster := ""
+			installation := ""
+			name := ""
+			organization := ""
+			stackName := ""
+
+			fmt.Printf("\n")
+			fmt.Printf("\n")
+			fmt.Printf("\n")
+			for _, filter := range filters {
+				fmt.Printf("%#v\n", filter.String())
+			}
+			fmt.Printf("gaugeValue: %#v\n", gaugeValue)
+			fmt.Printf("old gaugeValue: %#v\n", float64(1))
+			fmt.Printf("c.installationName: %#v\n", c.installationName)
+			fmt.Printf("vpc.ID: %#v\n", *vpc.VpcId)
+			for _, tag := range vpc.Tags {
+				fmt.Printf("tag: %#v\n", tag)
+				if *tag.Key == ClusterTag {
+					cluster = *tag.Value
+				}
+				if *tag.Key == InstallationTag {
+					installation = *tag.Value
+				}
+				if *tag.Key == NameTag {
+					name = *tag.Value
+				}
+				if *tag.Key == OrganizationTag {
+					organization = *tag.Value
+				}
+				if *tag.Key == StackNameTag {
+					stackName = *tag.Value
+				}
+			}
+
+			ch <- prometheus.MustNewConstMetric(
+				vpcs,
+				prometheus.GaugeValue,
+				gaugeValue,
+				*vpc.CidrBlock,
+				cluster,
+				*vpc.VpcId,
+				installation,
+				name,
+				organization,
+				stackName,
+				*vpc.State,
+			)
+		}
+	}
+
+	{
+		filters := []*ec2.Filter{
 			{
 				Name: aws.String(fmt.Sprintf("tag:%s", tagKeyInstallation)),
 				Values: []*string{
 					aws.String(c.installationName),
 				},
 			},
-		},
-	}
-	o, err := c.awsClients.EC2.DescribeVpcs(i)
-	if err != nil {
-		c.logger.Log("level", "error", "message", "could not list vpcs", "stack", fmt.Sprintf("%#v", err))
+		}
+		f(filters)
```

Comment:
For me it would clearer to have `[][]*ec2.Filter` and iterate over that instead of defining and calling `f`.